### PR TITLE
chore: Update build system to latest ocular: typescript+esbuild

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Background
- Update to latest ocular
#### Change List
- Update config files etc.

### CODE REVIEW

File: modules/constants/package.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
   ],
   "sideEffects": false,
   "scripts": {
-    "build-bundle": "ocular-bundle ./src/bundle.ts",
-    "pre-build": "npm run build-bundle && npm run build-bundle -- --env=dev"
+    "build-minified-bundle": "ocular-bundle ./src/bundle.ts -output=dist/dist.min.js",
+    "build-dev-bundle": "ocular-bundle ./src/bundle.ts -output=dist/dist.dev.js ----env=dev",
```
Comment (Pessimistress) [inline modules/constants/package.json:39]: You only need two dashes here

File: modules/engine/package.json [hunk 2]
```diff
@@ -35,14 +35,15 @@
...
-    "pre-build": "npm run build-bundle && npm run build-bundle -- --env=dev"
+    "build-minified-bundle": "ocular-bundle ./src/index.ts -output=dist/dist.min.js",
+    "build-dev-bundle": "ocular-bundle ./src/index.ts -output=dist/dist.dev.js ----env=dev",
+    "prepublishOnly": "npm run build-minified-bundle && npm run build-dev-=bundle"
   },
-  "dependencies": {
-    "@babel/runtime": "^7.0.0",
-    "@luma.gl/constants": "9.0.0-beta.4",
+  "peerDependencies": {
     "@luma.gl/core": "9.0.0-beta.4",
```
Comment (Pessimistress) [inline modules/engine/package.json:43]: Do not lock the peer dependency versions. I.e. `^9.0.0`. These are not managed by lerna during publish.

File: modules/constants/package.json [hunk 3]
```diff
@@ -35,8 +35,9 @@
...
+    "prepublishOnly": "npm run build-minified-bundle && npm run build-dev-=bundle"
```
Comment (Pessimistress) [inline modules/constants/package.json:40]: Typo

File: modules/engine/package.json [hunk 4]
```diff
@@ -35,14 +35,15 @@
...
+  },
+  "peerDependencies": {
+    "@luma.gl/core": "^9.0.0-beta.4",
+    "@luma.gl/shadertools": "^9.0.0-beta.4"
   },
   "dependencies": {
-    "@babel/runtime": "^7.0.0",
-    "@luma.gl/constants": "9.0.0-beta.4",
-    "@luma.gl/core": "9.0.0-beta.4",
-    "@luma.gl/shadertools": "9.0.0-beta.4",
```
Comment (donmccurdy) [inline modules/engine/package.json:45]: Question – When something is a `peerDependency` of a package, I'd generally expect to see it as a `devDependency` of either the package itself, or the workspace root if workspaces are involved. Is it OK and intended that `@luma.gl/*` peer dependencies appear in neither of those places?

**Review by donmccurdy — APPROVED** [review]

File: modules/engine/package.json (hunk 4 above)
Comment (ibgreen) [reply modules/engine/package.json:45]: You may be right. Perhaps it is the fact that the examples are part of the workspace that gets these installed right now.

That should probably be addressed for the long term. I can tinker with that after this lands, would need to remove examples make sure an install failure is triggered and then add these in appropriate places.

Comment (donmccurdy) [reply modules/engine/package.json:45]: I'm fine with whatever you decide on this! Just wasn't sure whether some build tools might expect to see dependencies in one of those places. Maybe something to leave as-is for now and revisit if we run into build issues.

**Pull request merged**